### ISSUE TITLE
Require Pigeon to verify message already included in block before sending next message

### ISSUE THREAD

**Issue by taariq** [VolumeFi/paloma#149]
# Background
Pigeon is sending messages too quickly for Paloma to process. As such, Paloma is unable to come to consensus on adding attestation and other chain information. We want Pigeon to check messages are submitted to Paloma and included in a block before sending another message.

# Done when

- [x] Pigeon Protos are all verified as sent/included in a block before the next message is sent.
- [x] Pigeon will not sign and broadcast more than one message per key per block
- [x] Pigeon will have a second key for singing messages that

**Comment by taariq** [comment]
@byte-bandit is this already resolved?

**Comment by maharifu** [comment]
We already wait for messages to be included in a block.

We're using the legacy lens-based client, that uses [broadcastTx](https://github.com/palomachain/pigeon/blob/master/util/ion/broadcast.go#L58) to communicate with paloma, which calls `BroadcastTxSync` and then polls `Tx` until the transaction is confirmed. I added a delay in paloma, which increased the time we spent polling.

Also, the lock in the wrappers [SendMsg](https://github.com/palomachain/pigeon/blob/master/chain/paloma/wrappers.go#L74) means we wait until one message is fully processed before sending the next one.
We could optimize this, but the client is still highly integrated with a single key, e.g. [here](https://github.com/palomachain/pigeon/blob/master/util/ion/tx.go#L150), and is not prepared for concurrent requests. I tried changing some of those to a dynamic key but still had issues, so we would need a larger effort to refactor the client.
